astropy/io/fits/hdu/compressed.py

```diff
@@ -672,6 +675,57 @@ def __init__(self, data=None, header=None, name=None,
         self._orig_bscale = self._bscale
         self._orig_bitpix = self._bitpix
 
+    def _remove_unnecessary_default_extnames(self, header):
+        """Remove default EXTNAME values if they are unnecessary.
+
+        Some data files (eg from CFHT) can have the default EXTNAME and
+        an explicit value.  This method removes the default if a more
+        specific header exists. It also removes any duplicate default
+        values.
+        """
+        if 'EXTNAME' in header:
+            indices = header._keyword_indices['EXTNAME']
+            # Only continue if there is more than one found
+            n_extname = len(indices)
+            if n_extname > 1:
+                extnames_to_remove = []
+                for index in indices:
+                    if header[index] == self._default_name:
+                        extnames_to_remove.append(index)
+                if len(extnames_to_remove) == n_extname:
+                    # Keep the first (they are all the same)
+                    extnames_to_remove.pop(0)
+                # Remove them all in reverse order to keep the index unchanged.
+                for index in reversed(sorted(extnames_to_remove)):
+                    del header[index]
+
+    @property
+    def name(self):
+        # Convert the value to a string to be flexible in some pathological
+        # cases (see ticket #96)
+        # Similar to base class but uses .header rather than ._header
+        return str(self.header.get('EXTNAME', self._default_name))
+
+    @name.setter
+    def name(self, value):
+        # This is a copy of the base class but using .header instead
+        # of ._header to ensure that the name stays in sync.
+        if not isinstance(value, str):
+            raise TypeError("'name' attribute must be a string")
+        if not conf.extension_name_case_sensitive:
+            value = value.upper()
+        if 'EXTNAME' in self.header:
+            # Expectation is that all EXTNAME headers found will be
+            # replaced in both _header and _image_header.
+            # Up to this point _header might be the original header
+            # but now we can call the code to remove unnecessary
+            # COMPRESSED_IMAGE EXTNAME since we know we are setting a
+            # new value.
+            self._remove_unnecessary_default_extnames(self._header)
```

Comment:
I think this could be removed (see below).

Reply:
Removed.